app/components/oxql-metrics/util.ts

```diff
@@ -122,29 +122,48 @@ export const sumValues = (timeseries: Timeseries[], arrLen: number): (number | n
     )
   )
 
+export const countValues = (timeseries: Timeseries[], arrLen: number): number[] =>
+  Array.from({ length: arrLen }).map((_, i) =>
+    R.pipe(
+      timeseries,
+      // get point at that index for each timeseries
+      R.map((ts) => ts.points.values.at(0)?.values.values?.[i]),
+      // filter out nulls (undefined shouldn't happen)
+      R.filter((p) => typeof p === 'number'),
+      // count non-null elements per series
+      (points) => points.length
+    )
+  )
+
 // Take the OxQL Query Result and return the data in a format that the chart can use
 // We'll do this by creating two arrays: one for the timestamps and one for the values
 // We'll then combine these into an array of objects, each with a timestamp and a value
 // Note that this data will need to be processed further, based on the kind of chart we're creating
 export const composeOxqlData = (data: OxqlQueryResult | undefined) => {
   let timeseriesCount = 0
-  if (!data) return { chartData: [], timeseriesCount }
+  if (!data) return { chartData: [], valueCounts: [] }
   const timeseriesData = data.tables[0].timeseries
   timeseriesCount = timeseriesData.length
-  if (!timeseriesCount) return { chartData: [], timeseriesCount }
-  // Extract timestamps (all series should have the same timestamps)
-  const timestamps =
-    timeseriesData[0]?.points.timestamps.map((t) => new Date(t).getTime()) || []
-  // Sum up the values across all time series
+  if (!timeseriesCount) return { chartData: [], valueCounts: [] }
+  // Extract timestamps. Series may have different timestamps, e.g. when new
+  // resources are created within the query interval
+  const timestamps = R.pipe(
+    timeseriesData,
+    R.map((series) => series.points.timestamps.map((t) => new Date(t).getTime())),
+    R.flat(),
```

Comment:
Not that perf matters a ton here, but `R.flatMap` "is identical to a [map](https://remedajs.com/docs/#map) followed by a [flat](https://remedajs.com/docs/#flat) of depth 1 (`flat(map(data, ...args))`), but slightly more efficient than calling those two methods separately"